perf(language_server): avoid creating HashMap in `Backend::did_change_watched_files`

Sysix opened the pull request:
The URI is only valid for one worker, so use a `Vec` to avoid transforming into a `Vec` later.

camc314 commented:
### Merge activity

* **Oct 17, 7:49 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 17, 7:49 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).
* **Oct 17, 7:53 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
perf(language_server): avoid creating HashMap in `Backend::did_change_watched_files` (#14735)

The URI is only valid for one worker, so use a `Vec` to avoid transforming into a `Vec` later.
